### PR TITLE
Update codebook documentation and harmonize method names

### PR DESCRIPTION
**Pull request by ambrosejcarr**
* update documentation
* `codebook.empty_codebook` --> `codebook.empty`
* `codebook._create_codebook` --> `codebook.from_numpy`
* `codebook.from_json` --> `codebook.open_json`
* `codebook.metric_decode` --> `codebook.decode_metric`  (enable tab completion of codebook.decode_*)

Other changes are refactorings that relate to the above changes.

### CODE REVIEW

File: starfish/codebook/codebook.py [hunk 1]
```diff
@@ -91,34 +68,37 @@ def code_length(self) -> int:
...
-                [0, 0]],
-
-               [[0, 0],
-                [0, 0],
-                [0, 0]]], dtype=uint8)
-        Coordinates:
-          * target     (target) object 'ACTA' 'ACTB'
-          * c          (c) int64 0 1 2
-          * r          (r) int64 0 1
+        Build an empty 3-round 2-channel codebook::
```
Comment (neuromusic) [inline starfish/codebook/codebook.py]: example shows a 2-round, 3-channel codebook

File: starfish/codebook/codebook.py [hunk 2]
```diff
@@ -127,17 +107,17 @@ def _empty_codebook(cls, code_names: Sequence[str], n_channel: int, n_round: int
 
         """
         data = np.zeros((len(code_names), n_channel, n_round), dtype=np.uint8)
-        return cls._create_codebook(code_names, n_channel, n_round, data)
+        return cls.from_numpy(code_names, n_channel, n_round, data)
 
     @classmethod
-    def _create_codebook(
+    def from_numpy(
```
Comment (neuromusic) [inline starfish/codebook/codebook.py]: nice!

File: starfish/codebook/codebook.py [hunk 3]
```diff
@@ -152,22 +132,28 @@ def _create_codebook(
...
-                [0, 0]],
-
-               [[0, 0],
-                [0, 0],
-                [0, 0]]], dtype=uint8)
-        Coordinates:
-          * target     (target) object 'ACTA' 'ACTB'
-          * c          (c) int64 0 1 2
-          * r          (r) int64 0 1
+        build a 3-round 2-channel codebook where :code:`ACTA` is specified by intensity in round 0,
```
Comment (neuromusic) [inline starfish/codebook/codebook.py]: example shows 2-round, 3-channel

File: starfish/codebook/codebook.py [hunk 4]
```diff
@@ -91,34 +68,37 @@ def code_length(self) -> int:
         return int(np.dot(*self.shape[1:]))
 
     @classmethod
-    def _empty_codebook(cls, code_names: Sequence[str], n_channel: int, n_round: int):
-        """create an empty codebook of shape (code_names, n_channel, n_round)
+    def empty(cls, code_names: Sequence[str], n_channel: int, n_round: int):
```
Comment (neuromusic) [inline starfish/codebook/codebook.py]: I would expect `.empty()` to work like numpy and initialize with whatever values are in memory. May be out of scope for this PR, but would `.zeroes()` be a better method call here?

**Comment by ttung** [comment]
ditto here -- would prefer `from_json`

**Pull request merged**